packages/sil-governance/src/routes/create-proposal/components/FormOptions/AmendCommitteeQuorum.tsx

```diff
@@ -1,52 +1,72 @@
+import { ChainId } from "@iov/bcp";
+import { ElectionRule, electionRuleIdToAddress } from "@iov/bns";
 import { FieldValidator, FormApi } from "final-form";
 import {
   Block,
+  composeValidators,
   FieldInputValue,
   SelectField,
   SelectFieldItem,
   TextField,
   Typography,
 } from "medulas-react-components";
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import * as ReactRedux from "react-redux";
 
+import { getConfig } from "../../../../config";
 import { RootState } from "../../../../store/reducers";
-import { getElectionRules } from "../ProposalForm";
+import { getAllElectionRules } from "../ProposalForm";
 import { isFraction } from ".";
 
 export const COMMITTEE_QUORUM_FIELD = "Committee Rule to amend";
 const COMMITTEE_QUORUM_INITIAL = "Select a rule";
 export const QUORUM_FIELD = "Quorum";
 const QUORUM_PLACEHOLDER = "2/3";
 
+const committeeIsSet = (value: FieldInputValue): string | undefined => {
+  if (value === COMMITTEE_QUORUM_INITIAL) return "Must select a rule";
+  return undefined;
+};
+
 interface Props {
   readonly form: FormApi;
+  readonly electionRule: ElectionRule | undefined;
 }
 
-const AmendCommitteeQuorum = ({ form }: Props): JSX.Element => {
+const AmendCommitteeQuorum = ({ form, electionRule }: Props): JSX.Element => {
   const governor = ReactRedux.useSelector((state: RootState) => state.extension.governor);
   const [ruleItems, setRuleItems] = useState<SelectFieldItem[]>([]);
 
   useEffect(() => {
     const updateCommitteeItems = async (): Promise<void> => {
+      const chainId = (await getConfig()).bnsChain.chainSpec.chainId as ChainId;
+      const electionRuleAddress = electionRule ? electionRuleIdToAddress(chainId, electionRule.id) : null;
+
       if (!governor) throw new Error("Governor not set in store. This is a bug.");
-      const electionRules = await getElectionRules(governor);
-      const ruleItems = electionRules.map(rule => {
+      const allRules = await getAllElectionRules(governor);
+      const writeableRules = allRules.filter(rule => rule.admin === electionRuleAddress);
+      const ruleItems = writeableRules.map(rule => {
         return {
           name: `${rule.id}: ${rule.title}`,
         };
       });
 
       setRuleItems(ruleItems);
```

Comment:
This can cause snapshot/DOM tests warnings if you will not implement unmount/cleanup code in `useEffect`

Reply:
Thanks for feedback. There are indeed multiple issues with this kind of async data loading, one is that the `setRuleItems(ruleItems)` can can cause an unhandles promise rejection. With the cleanup approach I was also having massive issues in the wallet (don't remember what exactly). The long term solution is probably to avoid async data loading in the UI components entirely and work with redux instead. However, this requires a big refactoring, which is out of scope here.